File: ios/BlePaymentKit/Sources/BlePaymentKit/SignalValidator.swift
import Foundation

public struct SignalValidator: Sendable {
    public init() {}

    public func finalRssi(rssi: Int, rssiDelta: Int) -> Int { rssi - rssiDelta }

    public func validate(rssi: Int, rssiDelta: Int, config: BlePaymentConfig) -> ValidationResult {
        finalRssi(rssi: rssi, rssiDelta: rssiDelta) >= config.rssiThreshold ? .valid : .invalid(.signalBelowThreshold)
    }
}

File: kit/android/ble-payment-kit/src/main/kotlin/ru/paymentguide/blepaymentkit/SignalValidator.kt
package ru.paymentguide.blepaymentkit

class SignalValidator {
    fun finalRssi(rssi: Int, rssiDelta: Int): Int = rssi - rssiDelta

    fun validate(rssi: Int, rssiDelta: Int, config: BlePaymentConfig): ValidationResult =
        if (finalRssi(rssi, rssiDelta) >= config.rssiThreshold) ValidationResult.Valid
        else ValidationResult.Invalid(BlePacketRejectReason.signalBelowThreshold)
}
